docs/data/date-pickers/rfc-field.md

```diff
@@ -0,0 +1,426 @@
+import { Picker } from '@base-ui-components/react-x-date-pickers/picker';
+
+function CustomDatePicker(props) {
+  const manager = useDateManager();
+
+  return (
+    <Popover.Root>
+      <Picker.Root manager={manager} {...props}>
+        <PickerField.Root manager={manager} {...props}>
+          <PickerField.Content>{/** See demo above */}</PickerField.Content>
+          <Popover.Trigger>📅</Popover.Trigger>
+        </PickerField.Root>
+        <Popover.Backdrop />
+        <Popover.Positioner>{/** See picker documentation */}</Popover.Positioner>
+      </Picker.Root>
+    </Popover.Root>
+  );
+}
+```
+
+:::success
+When doing that, the `<PickerField.Root />` component doesn't have to receive props like `minDate` because they are accessed using `usePickerContext()` (or maybe a dedicated `usePickerValidationContext()`). This makes composition viable between the picker and the field (same for the views).
+:::
+
+### With Material UI
+
+:::success
+No DX change here compared to today.
+The only change is that the field component are detecting if there is a picker around them and adding an opening button if so (instead of having the picker pass a prop to define this opening button).
+:::
+
+The field exposed by `@mui/x-date-pickers` and `@mui/x-date-pickers` automatically
+
+```tsx
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+
+<DatePicker slots={{ field: DateField }} />;
+```
+
+:::success
+The concept of slots does not fit this use case very well, but the exploration of a better DX to override part of the UI in self-contained component is outside the scope of this documentation, so this RFC uses the tools currently available.
+:::
+
+## Basic usage (multi input picker)
+
+### Without Material UI
+
+:::warning
+This is not planned for now.
+`Picker.*` could contain some additional components that would be the building blocks for `<MultiInputRangeField />` but it does not seem to be the priority for now.
+:::
+
+### With Material UI
+
+The `@mui/x-date-pickers` package also exposes a component to create multi input range fields as follow:
+
+```tsx
+import { useDateRangeManager } from '@mui/x-date-pickers/managers';
+import { MultiInputRangeField } from '@mui/x-date-pickers/MultiInputRangeField';
+
+function MultiInputDateTimeRangeField(props: MultiInputDateTimeRangeFieldProps) {
+  const manager = useDateTimeRangeManager(props);
+
+  return <MultiInputRangeField {...props} manager={manager} ref={ref} />;
+}
+```
+
+When used inside a picker, `<MultiInputRangeField />` can be passed directly and uses the `manager` exposed by `usePickerContext`:
+
+```tsx
+<DatePicker slots={{ field: MultiInputRangeField }} />
+```
+
+:::success
+There is a POC of this in [#15505](https://github.com/mui/mui-x/pull/15505).
+:::
+
+## Clearable field
+
+### Without Material UI
+
+The user can use the `<PickerField.Clear />` component to add a button to clear the value:
```

Comment:
I am not sure about the naming of this. Maybe:
```suggestion
The user can use the `<PickerField.ClearButton />` component to add a button to clear the value:
```
It's a nitpick, and every component that triggers some sort of action should probably have a consistent naming, so feel free to ignore my comment if it conflicts with what you have planned for other parts of the code.

Reply:
The Base UI team used `<Menu.Trigger />` not `<Menu.TriggerButton />`, same for `<Menu.Increment />`.
So I don't think adding "Button" would fit nicely.

But maybe `<PickerFIeld.Clear />` is too vague and short and `<PickerField.ClearValue />` would be better (or something else).
Clearly the naming of the buttons seems to be one of the hot spot of the new DX :laughing: